### ISSUE TITLE
Vaccination Details Screen: Firstname Surname in wrong order

### ISSUE THREAD

**Issue by vaubaehn** [corona-warn-app/cwa-app-android#3692]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [ ] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation) - I don't know
* [x] Bug is not already reported in another issue

## Technical details

- Device name: LG G4 (LGH-815)
- Android version: 6.0
- App version: 2.5.1

## Describe the bug
In the Vaccination Details Screen, detailed information about person and type of vaccination is displayed.
For each entry, the first line displays the type of data in German and English, the second line displays the data.

For the DCC holder's name,
first line is: "Name, Vorname / Name, First Name"
second line is: "Vorname Name"
Hence, firstname and surname are swapped in second line according to the first line.

<img src="https://user-images.githubusercontent.com/68376758/125309357-fb108a80-e331-11eb-80fa-fcb3332c0dfd.png" width="250">

## Steps to reproduce the issue
1. Import a Vaccination DCC
2. Go to Vacc DCC details screen
3. See firstname and surname swapped in second line according to information in first line.

## Expected behaviour

When first line is
"Name, Vorname / Name, First Name"
second line should be
"Name, Vorname"

## Possible Fix

Combine strings accordingly.

## Additional context

Please check, if this should also be aligned with Recovery Certificate and Test Certificate

----
Internal Tracking-ID: [EXPOSUREAPP-8393](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8393) duplicate of
Internal Tracking ID: [EXPOSUREAPP-7824](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7824)

**Comment by MikeMcC399** [comment]
@vaubaehn 
This is a regression from 2.4.3 which did not display the header
Name, Vorname / Name, First Name
above the actual name from a vaccination certificate.

CovPass 1.28.3 uses the same headers and displays the name correctly according to the headers i.e. in the order
Name, First Name

**Comment by vaubaehn** [comment]
@MikeMcC399 
Thanks! I was not sure whether the headers were also displayed in 2.4.3.

**Comment by heinezen** [comment]
@vaubaehn 

Thanks for the report. I've raised this issue internally.

----
Corona-Warn-App Open Source Team

**Comment by dsarkar** [comment]
@vaubaehn In your case, is the DoB shown in the format YYYY-MM-DD, both in DE and EN locale? Thanks.

**Comment by vaubaehn** [comment]
@dsarkar 

Hi Dipankar,
> In your case, is the DoB shown in the format YYYY-MM-DD, both in DE and EN locale? Thanks.

Are you referring to #3690 ?

I set different locales and compared DE/EN (US)/EN (UK)/TR. For all locales, DoB is in the format YYYY-MM-DD in CWA 2.5.1 in
a) Certificate Overview Screen - "active user card" where you select the active user whose DCCs will be displayed on top of the list
b) Certificate Details Screen - where the details of the DCCs are revealed (headers in German/English)

Imho, in 
a) the DoB **could** be changed DD.MM.YYYY, though for my personal taste it's not a must. I didn't look deeply into the code, but I think #3690 is fixing it for CWA 2.6.x. And in
b) the DoB **should** stay in the format YYYY-MM-DD, because, related to some anecdotal reports I gathered, verifier apps are rarely used in the wild, and gate keepers (even including **border control staff** in Spain and France) are directly looking into the holder's app and just checking for face validity. In these situations it's more helpful when DoB stays in a well-known international format.

For the case of a) DoB on "active user card" in the Certificate Overview Screen, I found the German "geboren" in lower case (in line with all other second lines in this screen), while in other locales (EN/TR) it was "Born" resp. "Doğum tarihi" with first letter uppercase while all other 2nd lines started with first letter lowercase. This is probably due to translation process and only a cosmetic issue, but you may want to adapt these, too. I hope I won't need to open an issue for that... /cc @harambasicluka 

Dear Dipankar, I hope you don't mind when I don't test locales in the future on my phone (2015 released LG G4 with Android 6): changing the locale results in many ANRs (but you could also say: device not responding...), unexpected reboots, shocking moments when Android tells you "this application is not installed" from apps that you just used hours before, downloads of several additional language packs (at least for Google apps) into an already tiny free internal storage space....
It was an interesting experience, but I'd like to avoid that in future - thanks for your understanding 😉 ❤️

**Comment by dsarkar** [comment]
Internal Tracking-ID: [EXPOSUREAPP-8393](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8393) duplicate of
Internal Tracking ID: [EXPOSUREAPP-7824](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7824)

**Comment by MikeMcC399** [comment]
Internal Tracking ID: [EXPOSUREAPP-7824](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7824)
is mentioned in 
- Test Certificate Details - First Name Last Name instead of other way around (EXPOSUREAPP-7824) #3826
- Move full name formatted to interface (EXPOSUREAPP-7824) #3832

of the [v2.7.0-RC0](https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.7.0-RC0) release candidate.

So please check if this issue should get the [Fix 2.7](https://github.com/corona-warn-app/cwa-app-android/labels/Fix%202.7) label applied to it.

I checked the vaccination name display in the details view of a vaccination certificate in 2.7.1-RC0 and it looks good.
(Name, First Name displayed in that order.)
 
I can't check the other types of certificate though.

It is at least partially fixed in 2.7 and hopefully fully fixed in this upcoming version.

**Comment by MikeMcC399** [comment]
@vaubaehn 
In CWA 2.7.1 Android, in the detailed display of a vaccination certificate I'm seeing

`Name, Vorname / Name, First Name`

shown in the correct order.

Maybe you'd like to check and close if you agree that your issue is fixed?

**Comment by vaubaehn** [comment]
@MikeMcC399 
As long as I want to continue to debug some business rules issues, I still need to create some DCCs (with invalid signature) for myself, and thus don't want to update to 2.7.1 yet.
But I'm trusting your findings and consider the issue to be solved, hence closing here.
In case anything is (still) wrong, I'm going to re-open this issue.

Thanks everybody involved here!

**Comment by MikeMcC399** [comment]
@vaubaehn 
Related to your comments in https://github.com/corona-warn-app/cwa-app-android/issues/3692#issuecomment-879879475 about formatting of date-of-birth, date formatting is planned to be harmonized in 2.9.x through https://github.com/corona-warn-app/cwa-app-android/pull/3987. (Of course if you are staying on an older version in order to test without being blocked by invalid signatures you won't benefit from it, but just to let you know.)
